wcfsetup/install/files/lib/acp/form/PackageUpdateServerAddForm.class.php

```diff
@@ -31,153 +41,176 @@ class PackageUpdateServerAddForm extends AbstractForm
     public $neededPermissions = ['admin.configuration.package.canEditServer'];
 
     /**
-     * server url
-     * @var string
-     */
-    public $serverURL = '';
-
-    /**
-     * server login username
-     * @var string
+     * @inheritDoc
      */
-    public $loginUsername = '';
+    public $objectActionClass = PackageUpdateServerAction::class;
 
     /**
-     * server login password
-     * @var string
+     * @inheritDoc
      */
-    public $loginPassword = '';
+    public $objectEditLinkController = PackageUpdateServerEditForm::class;
 
     /**
      * @inheritDoc
      */
-    public function readFormParameters()
+    protected function createForm()
     {
-        parent::readFormParameters();
-
-        if (isset($_POST['serverURL'])) {
-            $this->serverURL = StringUtil::trim($_POST['serverURL']);
-        }
-        if (isset($_POST['loginUsername'])) {
-            $this->loginUsername = $_POST['loginUsername'];
-        }
-        if (isset($_POST['loginPassword'])) {
-            $this->loginPassword = $_POST['loginPassword'];
+        parent::createForm();
+
+        $passwordPlaceholder = '';
+        if (
+            $this->formAction !== IFormDocument::FORM_MODE_CREATE
+            && $this->formObject instanceof PackageUpdateServer
+            && $this->formObject->loginPassword !== ''
```

Comment:
```suggestion
            && $this->formObject->loginUsername !== ''
```
Previously the username was checked and this should remain like this to not leak anything about the password.

Reply:
The processor `password` ensures that username and password will be set to an empty string in case no username is given while it removes the column's value when reading the object for editing it.
There should be no case a password is stored without a username. Also `password` shouldn't be empty in case a username is stored.